src/main/java/org/wildfly/security/auth/server/RealmIdentity.java

```diff
@@ -118,6 +119,29 @@ default Principal getRealmIdentityPrincipal() {
     boolean verifyEvidence(Evidence evidence) throws RealmUnavailableException;
 
     /**
+     * Determine if the given evidence can be trusted in lieu of verifying or acquiring a credential. This method is
+     * intended to be used for identity propagation purposes.
+     *
+     * @param evidence the evidence to check (must not be {@code null})
+     * @return {@code true} if the given evidence can be trusted, {@code false} otherwise
+     * @throws RealmUnavailableException if the realm is not able to handle requests for any reason
+     */
+    default boolean checkEvidenceTrusted(Evidence evidence) throws RealmUnavailableException {
```

Comment:
If we do this at the realm level, I'm worried this might lead to a lot of duplication and/or bizarre behavior.  In the discussions around the original concept I think it was discussed that the ServerAuthenticationContext could possibly make the determination of trust without talking to the realm.  Is there a reason that the realm has to be involved at all in this check?